pkg/controller/iotproject/iotproject_controller.go

```diff
@@ -261,9 +263,12 @@ func (r *ReconcileIoTProject) Reconcile(request reconcile.Request) (reconcile.Re
 		rc.Process(func() (result reconcile.Result, e error) {
 			return r.reconcileManaged(ctx, &request, project)
 		})
-		rc.Process(func() (result reconcile.Result, e error) {
-			return r.updateProjectStatus(ctx, project, rc.Error())
-		})
+		if !reflect.DeepEqual(project, original) {
+			reqLogger.Info("Project changed, updating status")
+			rc.Process(func() (result reconcile.Result, e error) {
```

Comment:
Shouldn't the change be made on:

~~~go
	// update status
	err := r.client.Status().Update(ctx, newProject)
~~~

There are multiple paths leading to a call to `updateProjectStatus` and now I don't think we would properly update the status any longer.

Reply:
@ctron yeah ... This is the first thing that moved the needle in the local testing ... I want to see how it affects tests ... And will see how to better refactor all this if effective